Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrElementBuilders.kt

```diff
@@ -57,6 +57,37 @@ internal class MacroConstantStubBuilder(
         }
         return listOf(declaration)
     }
+    }
+
+/*!
+Yet another mangler, particularly to avoid secondary clash, e.g. when a property in prototype (interface)
+is mangled and that will cause another clash in the class which implements this interface.
+Rationale: keep algorithm simple but use the mangling characters which are rare in normal code, and keep mangling easy readable.
+ */
+internal class Mangler (val names: Sequence<String> = emptySequence()) {
+    private val reserved = setOf("Companion")
+    private val mangledNames: MutableSet<String> = names.toMutableSet()
+    private val prefixBase = "\$"
+    fun mangle(name: String): String {
+        if (name in reserved) {
```

Comment:
The implementation below doesn't seem consistent with the rationale above.
It is overcomplicated. For example, there is probably no reason for deriving each next mangle from previous one.